app/controllers/cm_admin/resource_controller.rb

```diff
@@ -0,0 +1,203 @@
+module CmAdmin
+  class ResourceController < ApplicationController
+    include Pundit::Authorization
+    include Pagy::Backend
+
+    def cm_index(params)
+      @current_action = CmAdmin::Models::Action.find_by(@model, name: 'index')
+      # Based on the params the filter and pagination object to be set
+      @ar_object = filter_by(params, nil, @model.filter_params(params))
+      # resource_identifier
+      respond_to do |format|
+        if request.xhr?
+          format.html { render partial: '/cm_admin/main/table' }
+        else
+          format.html { render '/cm_admin/main/' + action_name }
+        end
+      end
+    end
+
+    def cm_show(params)
+      @current_action = CmAdmin::Models::Action.find_by(@model, name: 'show')
+      scoped_model = "CmAdmin::#{@model.name}Policy::Scope".constantize.new(Current.user, @model.name.constantize).resolve
+      @ar_object = scoped_model.find(params[:id])
+      resource_identifier
+      respond_to do |format|
+        format.html { render '/cm_admin/main/' + action_name }
+      end
+    end
+
+    def cm_new(params)
+      @current_action = CmAdmin::Models::Action.find_by(@model, name: 'new')
+      @ar_object = @model.ar_model.new
+      resource_identifier
+      respond_to do |format|
+        format.html { render '/cm_admin/main/' + action_name }
+      end
+    end
+
+    def cm_edit(params)
+      @current_action = CmAdmin::Models::Action.find_by(@model, name: 'edit')
+      @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
+      resource_identifier
+      respond_to do |format|
+        format.html { render '/cm_admin/main/' + action_name }
+      end
+    end
+
+    def cm_update(params)
+      @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
+      @ar_object.assign_attributes(resource_params(params))
+      resource_identifier
+      resource_responder
+    end
+
+    def cm_create(params)
+      @ar_object = @model.ar_model.name.classify.constantize.new(resource_params(params))
+      resource_identifier
+      resource_responder
+    end
+
+    def cm_destroy(params)
+      @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
+      respond_to do |format|
+        if @ar_object.destroy
+          format.html { redirect_back fallback_location: cm_admin.course_index_path, notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is successful" }
+        else
+          format.html { redirect_back fallback_location: cm_admin.course_index_path, notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is unsuccessful" }
+        end
+      end
```

Comment:
@anbublacky - We need to change the fallback location to the index page of the model which calls the action.
```suggestion
      @ar_object = @model.ar_model.name.classify.constantize.find(params[:id])
      redirect_url = CmAdmin::Engine.mount_path + "/#{@model.name.underscore.pluralize}/
      respond_to do |format|
        if @ar_object.destroy
          format.html { redirect_back fallback_location: redirect_url, notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is successful" }
        else
          format.html { redirect_back fallback_location: redirect_url, notice: "#{action_name.titleize} #{@ar_object.class.name.downcase} is unsuccessful" }
        end
      end
```